### ISSUE TITLE
Magento2.Annotation.MethodArguments.MethodArguments can't deal with phpstan-style array argument types

### ISSUE THREAD

**Issue by hostep** [magento/magento-coding-standard#276]
### Preconditions
This started happening after upgrading the coding-standard library from version 6 to version 10 on a custom module of mine.

I suspect that this library currently can't deal wit [phpstan-style array argument types](https://phpstan.org/writing-php-code/phpdoc-types#general-arrays)

### Steps to reproduce
1. Have a class like this:
```php
<?php

declare(strict_types=1);

namespace Whatever;

use Symfony\Component\Console\Output\OutputInterface;

class SomeClass
{
    /**
     * @param array<array<string, mixed>> $categoryData
     */
    public function outputResult(array $categoryData, OutputInterface $output): int
    {
        return 0;
    }
}
```
2. Run this library over it

### Expected result
1. No warnings

### Actual result
1. Warnings:
```
--------------------------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 2 WARNINGS AFFECTING 1 LINE
--------------------------------------------------------------------------------------------------------------------------------------
 14 | WARNING | @param is not found for one or more params in method annotation
    |         | (Magento2.Annotation.MethodArguments.MethodArguments)
 14 | WARNING | $categoryData parameter is missing in method annotation (Magento2.Annotation.MethodArguments.MethodArguments)
--------------------------------------------------------------------------------------------------------------------------------------
```

**Comment by shochdoerfer** [comment]
Just ran into the same issue. Would love to see this fixed. For now, I disabled the magento-coding-standard checks in our CI pipeline in favor of phpstan.
